src/client/auth.js

```diff
@@ -1,16 +1,67 @@
 const UUID = require('uuid-1345')
 const yggdrasil = require('yggdrasil')
+const fs = require('mz/fs')
+const promisify = require('es6-promisify')
+const mkdirp = promisify(require('mkdirp'))
```

Comment:
Node `mkdir` function now have a `recursive` option that does the same https://nodejs.org/api/fs.html#fs_fs_mkdir_path_options_callback
And `mkdirp` already return a promise https://www.npmjs.com/package/mkdirp#mkdirpdir-opts---promisestring--undefined